### ISSUE TITLE
Helpers are not getting called for can.Component if template is loaded through steal

### ISSUE THREAD

**Issue by opolyo01** [canjs/canjs#1808]
If you try loading template with steal from external stache file, the helpers will not be getting called
````
steal(
"./date-picker.stache!",
//global
"can",
function (template, can) {
    can.Component.extend({
        tag: "date-picker",
        template: template,
        scope: {},
        helpers:{
            modifiedName: function(name){
              name = typeof name === "function" ? name() : name;
              return "Custom " + name;
            }
        }
    });
}
);
````

However, if I declare template inline helpers is getting called
````
steal(
"./date-picker.stache!",
//global
"can",
function (template, can) {
    can.Component.extend({
        tag: "date-picker",
        template: "{{modifiedName 'randomName'}}",
        scope: {},
        helpers:{
            modifiedName: function(name){
              name = typeof name === "function" ? name() : name;
              return "Custom " + name;
            }
        }
    });
}
);
````

**Comment by justinbmeyer** [comment]
Does this break if the inline template is stache?

```
steal(
"./date-picker.stache!",
//global
"can",
function (template, can) {
    can.Component.extend({
        tag: "date-picker",
        template: can.stache("{{modifiedName 'randomName'}}"),
        scope: {},
        helpers:{
            modifiedName: function(name){
              name = typeof name === "function" ? name() : name;
              return "Custom " + name;
            }
        }
    });
}
);
```

**Comment by opolyo01** [comment]
Yes, it will break this way too.

**Comment by justinbmeyer** [comment]
So, I created the following test, which passes.  Can you confirm that if it works for you?

```
can.Component.extend({
    tag: "call-helper",
    template: can.stache("{{modifiedName 'randomName'}}"),
    scope: {},
    helpers:{
        modifiedName: function(name){
          name = typeof name === "function" ? name() : name;
          return "Custom " + name;
        }
    }
});

var frag = can.stache('<call-helper></call-helper>')();
console.log(frag.firstChild.firstChild.nodeValue);
```

What version of CanJS are you using?  I think this works in the latest versions.

**Comment by justinbmeyer** [comment]
The actual test in case someone needs it:

```
	test('template helpers passed to template (#1808)', function(){
		
		can.Component.extend({
	        tag: "call-helper",
	        template: can.stache("{{modifiedName 'randomName'}}"),
	        scope: {},
	        helpers:{
	            modifiedName: function(name){
	              name = typeof name === "function" ? name() : name;
	              return "Custom " + name;
	            }
	        }
	    });
	    
	    var frag = can.stache('<call-helper></call-helper>')();
	    equal(frag.firstChild.firstChild.nodeValue, "Custom randomName");
	    
	});
```

**Comment by opolyo01** [comment]
Yes, your tests passing because you don't pre-render a component into the DOM.  If instead of creating a fragment, you would do this --
```
// var frag = can.stache('<call-helper></call-helper>')();
  // console.log(frag.firstChild.firstChild.nodeValue);
  $("#out").html(can.view("appMustache", {}));
```
Where template is loaded from the .html file
```
<script id="appMustache" type="text/mustache">
  <call-helper></call-helper>
</script>
```

On the side note I realized that I cannot steal('can' on its own to get access to can/stache, I also need to load a stache file.  Not sure if it has anything to do with an issue.

**Comment by justinbmeyer** [comment]
What do you mean by pre-render a component into the DOM?

**Comment by opolyo01** [comment]
$("#out").html(can.view("appMustache", {}));
instead of 
var frag = can.stache('<call-helper></call-helper>')();

**Comment by gmahomarf** [comment]
Has there been any progress on this issue? I'm having a similar issue, but I'm using RequireJS instead of Steal. If I register the helpers with `can.stache.registerHelper`, everything works fine. If I add them to the `helpers` property of a new `Component`, they don't work.

**Comment by pYr0x** [comment]
Do you mixing mustache and stache templates?

If you using stache all parent components and templates have to be a stache-template

**Comment by gmahomarf** [comment]
@pYr0x I'm using stache only, but what does that have to do with anything? 
From docs ([link](http://canjs.com/docs/can.stache.html#section_Differencesfromcan_mustache)):
>Differences from can.mustache
>can.stache is largely compatable with can.mustache. There are three main differences:

>- Passes values in the scope to can.Component with {key}.
>- section renderers return documentFragments.
>- Element callbacks like {{(el) -> CODE}} are no longer supported.

I'm not passing values to any child components, I'm not using Element callbacks, and I assume section rendering works fine, because the section rendered by the helper I'm using displays something (always defaults to truthy, so it's not working entirely as expected, but it's rendering nonetheless). I don't see how that would affect my code

**Comment by justinbmeyer** [comment]
I still need a breaking test case because I'm not sure what the problem is.

**Comment by gmahomarf** [comment]
@justinbmeyer alright. I'll see if I can create a test case to reproduce the issue.

**Comment by justinbmeyer** [comment]
Here's a jsbin that points to latest: http://jsbin.com/tunutin/edit?html,js,output

**Comment by pYr0x** [comment]
@gmahomarf: it's not about the syntax. it's about you declaring the template
this example shows you the problem:

http://jsbin.com/jusovafeva/edit?html,js,console,output

replace "stacheTemplate" with "mustacheTemplate" and you will notice that the helper in child-comp will call with the with the mustache-template, but not with the stache-template

**Comment by gmahomarf** [comment]
@justinbmeyer  I'm still working on getting you a test case that matches my current setup (maybe I screwed something up), but I think @pYr0x may be on to something there. After looking at his jsbin code, I decided to test something in my code. All my templates are in <view_name>.stache files and I set them to each Component's template with can.view(url). I renamed one of them to <view_name>.mustache, and the Component's helpers started working. Changed back to <view_name>.stache, broken.

**Comment by gmahomarf** [comment]
@justinbmeyer all my test cases pass, so I can't reproduce my issue outside my own code. I've got to get this project out, so for now I've worked around it by using require's text plugin to load the template as text and then set each Component's template to can.stache(requiredTemplate). As soon as I'm done with this project, I'll work on creating a test that fails the same way my code is failing. Thanks
